src/Tasks/Microsoft.Common.CurrentVersion.targets

```diff
@@ -5366,6 +5385,37 @@ Copyright (C) Microsoft Corporation. All rights reserved.
 
   </Target>
 
+  <!--
+    ============================================================
+                                        _CopyDifferingSourceItemsToOutputDirectory
+
+    Copy files that have the CopyToOutputDirectory attribute set to 'IfDifferent'.
+    ============================================================
+    -->
+  <Target
+      Name="_CopyDifferingSourceItemsToOutputDirectory"
+      Condition=" '@(_SourceItemsToCopyToOutputDirectoryIfDifferent)' != '' ">
+
+    <!--
+        Using SkipUnchangedFiles="true" because we want only differing files.
+        -->
+    <Copy
```

Comment:
The `DoCopyIfNecessary` that is used by the copy task in this case is doing it's check by
`(SkipUnchangedFiles && IsMatchingSizeAndTimeStamp(sourceFileState, destinationFileState))`
If I understand this correctly, this is same for both PreserveNewest and IfDifferent(or at least I found no other check)
The way the IfDifferent properties above are set up also makes me think that IfDifferent behavior is identical to KeepNewest.
Is this correct or did I miss something please?
If it is correct, why do we introduce it?
If I am mistaken, what did I miss please?

Reply:
There is one slight difference - KeepNewest doesn't copy the file if it changes in the destination after the previous build.

That can lead to unfortunate behaviors - e.g. when you are deploying config file, or DB or something like that during your build, but can alter it as part of tests - then new build wont reset the state - and that forces ppl to use 'Always'.

There are couple other similar scenarios - that's why the 'IfDifferent' was introduced